lab_6_pipeline/pipeline.py

```diff
@@ -62,11 +111,21 @@ def __init__(
             corpus_manager (CorpusManager): CorpusManager instance
             analyzer (LibraryWrapper | None): Analyzer instance
         """
+        self._corpus = corpus_manager
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        docs = [article.text for article in self._corpus.get_articles().values()]
+        if self.analyzer:
+            docs = self.analyzer.analyze(docs)
+        for i, article in enumerate(self._corpus.get_articles().values()):
```

Comment:
zip with docs and iterate through all values simultaneously

Reply:
I'm not sure that I have got you right, like this? If not, I'll rework

Reply:
oh, no I think I got the idea about using zip...